.github/workflows/pages-gem.yml

```
@@ -0,0 +1,22 @@
+name: Publish Gem
+
+on:
+  release:
+    types: [released]
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.2'
+      - name: Build gem
+        run: |
+          gem build github-pages.gemspec
+      - name: Publish
+        run: |
+          gem push github-pages.gem --key ${{ secrets.PAGES_GEM_PUBLISH }}
```

1. Does a release here in `jekyll-commonmark-ghpages` warrant publishing `github-pages`?

2. Since there's no `github-pages.gemspec` here, only `jekyll-commonmark-ghpages.gemspec`, won't this fail? (Maybe I'm horribly missing something, sorry if that's the case…)

3. Bonus question, since the resulting gem has a version in its filename, you won't have `package-name.gem` but e. g. `package-name-0.4.0.gem` to push. Won't you need some more output magic to push the correct filename?

---

Thanks for pointing out that yeah this workflow will not work as is 😁. Will fix it when I get a chance for next release (we copied it in lots of repos actually!).

---

`actions/starter-workflows` has a neat trick [ci/gem-push.yml#L39-L48](https://github.com/actions/starter-workflows/blob/be552580a63fc68faa5036cf1ae2646d3ee1bb37/ci/gem-push.yml#L39-L48) if you don't want to manage gem file names in every repo separately… simply wildcarding it everywhere;)

```suggestion
          gem build *.gemspec
      - name: Publish
        run: |
          gem push *.gem --key ${{ secrets.PAGES_GEM_PUBLISH }}
```